src/pkg/bundle/deploy.go

```diff
@@ -541,3 +485,200 @@ func formFilePath(anchorPath string, filePath string) (string, error) {
 }
+
+// PreDeployValidation validates the bundle before deployment
+func (b *Bundle) PreDeployValidation() (string, string, string, error) {
+
+	// Check that provided oci source path is valid, and update it if it's missing the full path
+	source, err := CheckOCISourcePath(b.cfg.DeployOpts.Source)
+	if err != nil {
+		return "", "", "", err
+	}
+	b.cfg.DeployOpts.Source = source
+
+	// create a new provider
+	provider, err := NewBundleProvider(b.cfg.DeployOpts.Source, b.tmp)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// pull the bundle's metadata + sig
+	loaded, err := provider.LoadBundleMetadata()
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// validate the sig (if present)
+	if err := ValidateBundleSignature(loaded[config.BundleYAML], loaded[config.BundleYAMLSignature], b.cfg.DeployOpts.PublicKeyPath); err != nil {
+		return "", "", "", err
+	}
+
+	// read in file at config.BundleYAML
+	message.Debugf("Reading YAML at %s", loaded[config.BundleYAML])
+	bundleYAML, err := os.ReadFile(loaded[config.BundleYAML])
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// todo: we also read the SHAs from the uds-bundle.yaml here, should we refactor so that we use the bundle's root manifest?
+	if err := goyaml.Unmarshal(bundleYAML, &b.bundle); err != nil {
+		return "", "", "", err
+	}
+
+	// validate bundle's arch against cluster
+	err = ValidateArch(config.GetArch(b.bundle.Build.Architecture))
+	if err != nil {
+		return "", "", "", err
+	}
+
+	bundleName := b.bundle.Metadata.Name
+	return bundleName, string(bundleYAML), source, err
+}
+
+// ConfirmBundleDeploy prompts the user to confirm bundle creation
+func (b *Bundle) ConfirmBundleDeploy() (confirm bool) {
+
+	pkgviews := formPkgViews(b)
+
+	message.HeaderInfof("🎁 BUNDLE DEFINITION")
+	pterm.Println("kind: UDS Bundle")
+
+	message.HorizontalRule()
+
+	message.Title("Metatdata:", "information about this bundle")
+	utils.ColorPrintYAML(b.bundle.Metadata, nil, false)
+
+	message.HorizontalRule()
+
+	message.Title("Build:", "info about the machine, UDS version, and the user that created this bundle")
+	utils.ColorPrintYAML(b.bundle.Build, nil, false)
+
+	message.HorizontalRule()
+
+	message.Title("Packages:", "definition of packages this bundle deploys, including variable overrides")
+
+	for _, pkg := range pkgviews {
+		utils.ColorPrintYAML(pkg.meta, nil, false)
+		utils.ColorPrintYAML(pkg.overrides, nil, false)
+	}
+
+	message.HorizontalRule()
+
+	// Display prompt if not auto-confirmed
+	if config.CommonOptions.Confirm {
+		return config.CommonOptions.Confirm
+	}
+
+	prompt := &survey.Confirm{
+		Message: "Deploy this bundle?",
+	}
+
+	if err := survey.AskOne(prompt, &confirm); err != nil || !confirm {
+		return false
+	}
+	return true
+}
+
+type PkgView struct {
+	meta      map[string]string
+	overrides map[string]interface{}
+}
+
+// formPkgViews creates a unique pre deploy view of each packages set overrides and Zarf variables
+func formPkgViews(b *Bundle) []PkgView {
```

Comment:
Make sense to break this f'n up? Nearing a hundred lines, could help future us with readability :)